pkg/controller/storagecluster/placement.go

```diff
@@ -51,25 +55,39 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 	return placement
 }
 
-//convertLabelToNodeSelector returns NodeSelectorTerm type from a given LabelSelector
-func convertLabelToNodeSelector(labelSelector metav1.LabelSelector) corev1.NodeSelectorTerm {
-	term := corev1.NodeSelectorTerm{}
+//convertLabelToNodeSelectorRequirements returns a NodeSelectorRequirement list from a given LabelSelector
+func convertLabelToNodeSelectorRequirements(labelSelector metav1.LabelSelector) []corev1.NodeSelectorRequirement {
+	reqs := []corev1.NodeSelectorRequirement{}
 	for key, value := range labelSelector.MatchLabels {
 		req := corev1.NodeSelectorRequirement{}
 		req.Key = key
 		req.Operator = corev1.NodeSelectorOpIn
 		req.Values = append(req.Values, value)
-		term.MatchExpressions = append(term.MatchExpressions, req)
+		reqs = append(reqs, req)
 	}
 	numIter := len(labelSelector.MatchExpressions)
 	for i := 0; i < numIter; i++ {
 		req := corev1.NodeSelectorRequirement{}
 		req.Key = labelSelector.MatchExpressions[i].Key
 		req.Operator = corev1.NodeSelectorOperator(labelSelector.MatchExpressions[i].Operator)
 		req.Values = labelSelector.MatchExpressions[i].Values
-		term.MatchExpressions = append(term.MatchExpressions, req)
+		reqs = append(reqs, req)
+	}
+	return reqs
+}
+
+func appendNodeRequirements(placement *rookv1.Placement, reqs ...corev1.NodeSelectorRequirement) {
+	if placement.NodeAffinity == nil {
+		placement.NodeAffinity = &corev1.NodeAffinity{}
+	}
+	if placement.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		placement.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &corev1.NodeSelector{}
+	}
```

Comment:
if `placement.NodeAffinity == nil` then why do we need to check for `placement.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil`? It'll obviously be `nil`.

Reply:
I think this pattern is perfectly fine to avoid special cases and deep nesting. As long as this check covers the special case of (originally) `placement.NodeAffinity == nil` in which we have set `placement.NodeAffinity` specially.

Reply:
First you check for nil in the NodeAffinity, since you cannot guarantee it will be set. Similarly, Req... is a pointer, and that field is ALSO not guaranteed to be set (it would be a NodeAffinity with Preferred... set instead).